src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslCertificateFinder.cs

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed class OpenSslCertificateFinder : IFindPal
+    {
+        private readonly X509Certificate2Collection _findFrom;
+        private readonly X509Certificate2Collection _copyTo;
+        private readonly bool _validOnly;
+
+        internal OpenSslCertificateFinder(X509Certificate2Collection findFrom, X509Certificate2Collection copyTo, bool validOnly)
+        {
+            _findFrom = findFrom;
+            _copyTo = copyTo;
+            _validOnly = validOnly;
+        }
+
+        public string NormalizeOid(string maybeOid, OidGroup expectedGroup)
+        {
+            Oid oid = new Oid(maybeOid);
+
+            // If maybeOid is interpreted to be a FriendlyName, return the OID.
+            if (!StringComparer.OrdinalIgnoreCase.Equals(oid.Value, maybeOid))
+            {
+                return oid.Value;
+            }
+
+            FindPal.ValidateOidValue(maybeOid);
+            return maybeOid;
+        }
+
+        public void FindByThumbprint(byte[] thumbprint)
+        {
+            FindCore(cert => cert.GetCertHash().ContentsEqual(thumbprint));
```

Comment:
Again, may not matter at all, but this is going to allocate a closure/delegate due to capturing thumbprint.